src/node_sets.cpp

```diff
@@ -73,6 +77,21 @@ class NodeSets
         parse_compound(j, node_sets_);
     }
 
+    explicit NodeSets(const fs::path& path)
+        : NodeSets(json::parse(std::fstream(path))) {}
```

Comment:
Unlike `fromFile` this doesn't check that the file exists. Maybe something
```
const fs::path& validate_path(fs::path& path)
     if (!fs::exists(path)) {
        throw SonataError(fmt::format("Path does not exist: {}", std::string(path)));
    }
    return path;
}

explicit NodeSets(const fs::path& path)
    : NodeSets(json::parse(std::fstream(validate_path(path)))) {}


static std::unique_ptr<NodeSets> fromFile(const std::string& path_) {
    return std::make_unique<detail::NodeSets>(path);
}
```

Finally, probably pointless comment but: `json::parse` only needs an `std::ifstream`.